Code/GraphMol/Resonance.cpp

```diff
@@ -834,6 +834,7 @@ bool ConjElectrons::purgeMaps(CEMap &ceMap, CEDegCount &ceDegCount,
   bool ok = true;
   bool changed = true;
   while (changed) {
+    changed = false;
     for (auto it = ceMap.begin(); it != ceMap.end();) {
       if (!it->second->checkMetrics(ceStats, changed)) {
```

Comment:
checkMetrics changes the value of changed, but if ceMap is empty, this never gets called and changed remains true causing an infinite loop.

Reply:
Yes, of course, you are right. My bad - I should have engaged my brain before replying :/.